src/engine/threaded_engine.h

```diff
@@ -331,6 +332,10 @@ class ThreadedEngine : public Engine {
       kill_.store(true);
     }
     finished_cv_.notify_all();
+    // close opened libraries
+    for (auto const& lib : loaded_libs) {
+      close_lib(lib.second);
```

Comment:
why this in the engine? I suggest to do global cleanup in initialize.cc.

related: https://github.com/apache/incubator-mxnet/pull/15762

Reply:
addressed.

Reply:
addressed